cli/flox-rust-sdk/src/models/lockfile.rs

```diff
@@ -348,30 +352,22 @@ impl FromIterator<ResolutionFailure> for ResolutionFailures {
 }
 
 /// Data relevant for formatting a resolution failure
+///
+/// This may wrap messages returned from the catalog with additional information
+/// extracted from the manifest.
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
 pub enum ResolutionFailure {
```

Comment:
If we dont need the extra metadata for anything but messaging (in `format_single_resolution_failure`), i think the formatting could even be folded into the construction and ResolutionFailure wouldn't need to mirror the message variants:

```
struct ResolutionFailure {
    custom_message: Option<String>,
    source: ResolutionMessage
}
```

or (to keep custom messages)

```
enum ResolutionFailure 
{
	Derived {
        custom_message: Option<String>,
        source: ResolutionMessage
    },
    Custom {
        message: String
    } 
}
```

That being said, i think this solution is already an improvement, and not in the way f further reduction.

Reply:
We do partition using `PackageNotFound` so we'd have to handle that somehow

https://github.com/flox/flox/blob/aac8748cab5e08180930ba6a16a4624bf5a2996f/cli/flox/src/commands/install.rs#L233

Reply:
We can still partition through `source` then no?

Reply:
Sure we could